src/main/java/com/ericsson/ei/config/ConfigurationLogger.java

```diff
@@ -1,80 +1,104 @@
-                + "ldap.enabled: " + environment.getProperty("ldap.enabled") + "\n"
-                + "ldap.server.list: " + environment.getProperty("ldap.server.list") + "\n"
-                + "logging.level.root: " + environment.getProperty("logging.level.root") + "\n"
-                + "logging.level.org.springframework.web: " + environment.getProperty("logging.level.org.springframework.web") + "\n"
-                + "logging.level.com.ericsson.ei: " + environment.getProperty("logging.level.com.ericsson.ei") + "\n"
-                + "\nThese properties are only some of the configurations, more configurations could have been provided.\n");
+        List<MapPropertySource> propertySources = findApplicationProperties();
+        for (MapPropertySource propertySource : propertySources) {
+            String propertiesLogMessage = createPropertiesLogMessage(propertySource);
+            LOGGER.debug(propertiesLogMessage);
+        }
+    }
+
+    private List<MapPropertySource> findApplicationProperties() {
+        List<MapPropertySource> propertySources = new ArrayList<>();
+        for (PropertySource<?> propertySource : ((ConfigurableEnvironment) environment).getPropertySources()) {
+            if (propertySource.getName().contains("applicationConfig")) {
+                propertySources.add((MapPropertySource) propertySource);
+            }
+        }
+        return propertySources;
+    }
+
+    private String createPropertiesLogMessage(MapPropertySource propertySource) {
+        String[] propertyNames = propertySource.getPropertyNames();
+        Arrays.sort(propertyNames);
+        StringBuilder propertiesLogMessage = new StringBuilder();
+        propertiesLogMessage.append(String.format("\n##### %s #####\n", propertySource.getName()));
+        for (String propertyName : propertyNames) {
+            String propertyValue = getPropertyValueAndCheckForSpecialCases(propertyName);
+            propertiesLogMessage.append(String.format("%s: %s\n", propertyName, propertyValue));
+        }
+        return propertiesLogMessage.toString();
+    }
+
+    private String getPropertyValueAndCheckForSpecialCases(String propertyName) {
+        final String maskedProperty = "*****";
+        switch (propertyName) {
+        case "spring.data.mongodb.uri":
+            final String unsafeUri = environment.getProperty("spring.data.mongodb.uri");
+            final String safeUri = MongoUri.getUriWithHiddenPassword(unsafeUri);
+            return safeUri;
+        case "rabbitmq.password":
+            String rabbitPassword = environment.getProperty("rabbitmq.password");
+            return StringUtils.isEmpty(rabbitPassword) ? "" : maskedProperty;
+        case "spring.mail.password":
+            String mailPassword = environment.getProperty("spring.mail.password");
+            return StringUtils.isEmpty(mailPassword) ? "" : maskedProperty;
+        case "ldap.server.list":
```

Comment:
Why don´t you use SafeLdapServer.class from  ei utils?

Reply:
Didn't know it existed.